pkg/comp-functions/functions/vshn-postgres-func/mailgun_alerting.go

```diff
@@ -0,0 +1,127 @@
+package vshnpostgres
+
+import (
+	"context"
+	"fmt"
+
+	xkube "github.com/crossplane-contrib/provider-kubernetes/apis/object/v1alpha1"
+	alertmanagerv1alpha1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1"
+	vshnv1 "github.com/vshn/appcat/apis/vshn/v1"
+	runtime "github.com/vshn/appcat/pkg/comp-functions/runtime"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+)
+
+func MailgunAlerting(ctx context.Context, iof *runtime.Runtime) runtime.Result {
+	log := controllerruntime.LoggerFrom(ctx)
+	log.Info("Starting mailgun-alerting function")
+
+	comp := &vshnv1.VSHNPostgreSQL{}
+	err := iof.Observed.GetComposite(ctx, comp)
+	if err != nil {
+		return runtime.NewFatalErr(ctx, "Cannot get composite from function io", err)
+	}
+	// Wait for the next reconciliation in case instance namespace is missing
+	if comp.Status.InstanceNamespace == "" {
+		return runtime.NewWarning(ctx, "Composite is missing instance namespace, skipping transformation")
+	}
+
+	if comp.Spec.Parameters.Monitoring.Email != "" {
```

Comment:
(very small)Nitpick (Feel free to ignore): This would IMO be more readable if you did an early return, so instead do

```golang
if email == "" {
    return runtime.NewNormal()
}
```

(This is IMO a good article explaining why this helps readability https://maelvls.dev/go-happy-line-of-sight/)